Improve Linux installation instructions

ptgott opened the pull request:
Closes #32387
Closes #38144

In the Installation page, the instructions for installing Teleport on Linux reuses the partial we use for how-to guides, `install-linux.mdx`. This is inappropriate for an installation reference, since the partial is not intended for users interested in a specific package manager. This change provides a comprehensive description of our package repositories for Teleport installations on Linux.

Since the Installation page is a common entrypoint for users looking to get started with Teleport, this change also adds a section to the Linux installation instructions with recommended pages to follow for setting up Teleport.

Also, since we have a script available form the corporate Teleport site that automatically chooses a package manager and configures it to target a Teleport package repo, this change adds instructions to use this.

I have organized the Linux installation section to place H3 sections from less work/configuration to more work/configuration, from the one-line installation script to package manager-specific instructions and, finally, general information about our package repositories.

ptgott committed:
Improve Linux installation instructions

Closes #32387

In the Installation page, the instructions for installing Teleport on
Linux reuses the partial we use for how-to guides, `install-linux.mdx`.
This is inappropriate for an installation reference, since the partial
is not intended for users interested in a specific package manager. This
change provides a comprehensive description of our package repositories
for Teleport installations on Linux.

Since the Installation page is a common entrypoint for users looking to
get started with Teleport, this change also adds a section to the Linux
installation instructions with recommended pages to follow for setting
up Teleport.

Also, since we have a script available form the corporate Teleport site
that automatically chooses a package manager and configures it to target
a Teleport package repo, this change adds instructions to use this.

I have organized the Linux installation section to place H3 sections
from less work/configuration to more work/configuration, from the
one-line installation script to package manager-specific instructions
and, finally, general information about our package repositories.

r0mant commented on docs/pages/installation.mdx:
```diff
@@ -48,55 +48,391 @@ supports most features on Windows 10 and later.*
-- Ubuntu >= 16.04 (`ubuntu`)
-- Amazon Linux 2 and 2023 (`amzn`)
-- SLES >= 12 SP5, >= 15 SP5 (`sles`)
+If running Teleport Enterprise Cloud, Teleport agents must be running version
+(=cloud.version=) or one major version below.
```
I think we should remove this remark because on Cloud we want everyone to be using automatic upgrades.

ptgott replied:
Removing this paragraph and adding a more explicit recommendation for Cloud users to enable automatic agent upgrades.

r0mant commented on docs/pages/installation.mdx:
```diff
@@ -48,55 +48,391 @@ supports most features on Windows 10 and later.*
-`ID_LIKE` to install packages of the closest supported distribution.
+| Value        | Edition                           |
+|--------------|-----------------------------------|
+| `cloud`      | Teleport Enterprise Cloud         |
```
What does the script actually do when you specify "cloud" edition? We want all Cloud users to be using automatic upgrades, will this install the "teleport-ent-updater" as well?

ptgott replied:
It currently does not. I'll open an issue in the repo where the one-line installation script lives so we can add this functionality.

r0mant commented on docs/pages/installation.mdx:
```diff
@@ -48,55 +48,391 @@ supports most features on Windows 10 and later.*
+$ source /etc/os-release
+# Add the Teleport APT repository for v(=teleport.major_version=). You'll need to update this
+# file for each major release of Teleport.
+$ echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] \
```
These instructions are incorrect for Cloud. Cloud users should be using stable/cloud channel, and must also install teleport-ent-updater to be enrolled in automatic upgrades.

r0mant reviewed (changes requested):
The current instructions are inaccurate for Cloud users because they don't enroll users in automatic upgrades. I think we should direct all Cloud users (both Team and Enterprise) to https://goteleport.com/docs/choose-an-edition/teleport-cloud/downloads/.

ptgott committed:
Respond to r0mant feedback

- Remove the version guidance for Cloud users and make the
  recommendation for them to enable automatic agent upgrades more
  prominent.

- Add instructions for installing `teleport-ent-updater`.

- Add instructions for changing the Teleport version to `cloud` for
  Team/Enterprise Cloud in package manager examples. Reorganize the
  package manager instructions so there is a single tabbed example of
  setting the version. To account for this, refactor the subsections of
  the "Linux" section so there is a single H3 for TAR and a single H3
  for package repositories.

ptgott committed:
Respond to additional r0mant feedback

- Make it clear throughout the section that we recommend using the
  one-line installation script, which install the automatic updater for
  Teleport Cloud users.

- Condense package repository and TAR archive H3s. By simplifying these
  sections, we can make the instructions to use the updater on Cloud
  more prominent.

- Warn against using TAR archives with Teleport Cloud, and remove
  TabItems for these editions in the TAR archive section.

- Incorporate instructions introduced by #38401 to query the Proxy
  Service for the latest compatible version.

r0mant commented on docs/pages/installation.mdx:
```diff
@@ -48,56 +49,345 @@ supports most features on Windows 10 and later.*
-| Automatic Updates                                                              | systemd-based | 9+                       | 16.04+   | 7+          | 2, 2023                | 12 SP5, 15 SP5 |
-| Installation through apt/yum/zypper repos                                      | systemd-based | 9+                       | 16.04+   | 7+          | 2, 2023                | 12 SP5, 15 SP5 |
+If you are starting out with Teleport, we recommend beginning with a [Teleport
+Team](https://goteleport.com/signup/) account. From there, the only Teleport
```
Nit: I would say "Teleport Cloud" instead of "Teleport Team" to keep it more generic. We're introducing other plans soon.

```suggestion
Cloud](https://goteleport.com/signup/) account. From there, the only Teleport
```

r0mant commented on docs/pages/installation.mdx:
````diff
@@ -48,56 +49,345 @@ supports most features on Windows 10 and later.*
+Teleport:
+
+```code
+$ curl https://goteleport.com/static/install.sh | bash -s ${TELEPORT_VERSION?} ${TELEPORT_EDITION?} 
````
Does this script install auto-updater package when installing a cloud edition?

ptgott replied:
It does as of gravitational/next#2407

r0mant commented on docs/pages/installation.mdx:
```diff
@@ -48,56 +49,345 @@ supports most features on Windows 10 and later.*
+   archive:
+
+   <Tabs>
+   <TabItem label="Teleport Community Edition">
```
When reading through this I actually missed the "warning" above and was looking for a "Teleport Cloud" tab. What do you think about adding the "Teleport Cloud" tab here as well and adding the same warning text into it saying that Cloud users need to install using script or a package? I think it would draw their attention better than just a warning since they would (hopefully) click on the Cloud tab.

ptgott replied:
I found this a little tricky because the `Tabs` component only applies to assigning environment variables, not the whole procedure. I've removed the `Tabs` component and suggested a way to do this through regular body text so readers aren't tempted to skip the intro text.

r0mant commented on docs/pages/installation.mdx:
```diff
@@ -48,56 +49,345 @@ supports most features on Windows 10 and later.*
+   </TabItem>
+   </Tabs>
+
+### TAR archives
```
What do you think about adding this note to highlight even more this method is not suitable for Cloud?
```suggestion
### TAR archives (self-hosted only)
```

ptgott replied:
Sounds good!

r0mant reviewed (commented):
Sorry about the slow turnaround here @ptgott. I left a couple more thoughts but overall looks pretty good to me now. Once those get addressed, should be ready to :shipit:

ptgott committed:
Respond to r0mant feedback

- Remove Teleport Team mention
- Add "(self-hosted only)" to the TAR subsection
- Remove a Tabs component in the TAR section so readers aren't tempted
  to interact with the component before reading the initial body text.

ptgott committed:
Fix linter errors
